sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-batch-scoring/image-classification-multiclass-batch-scoring-non-mlflow-model.ipynb

```diff
@@ -0,0 +1,842 @@
+    "# create a batch deployment\n",
+    "deployment = BatchDeployment(\n",
+    "    name=\"non-mlflow-deployment\",\n",
+    "    description=\"this is a sample non-mlflow deployment\",\n",
+    "    endpoint_name=batch_endpoint_name,\n",
+    "    model=registered_model.id,\n",
+    "    code_path=\"./scripts\",\n",
+    "    scoring_script=\"batch_scoring.py\",\n",
+    "    environment=env,\n",
+    "    compute=\"cpu-cluster\",\n",
+    "    instance_count=2,\n",
+    "    max_concurrency_per_instance=2,\n",
+    "    mini_batch_size=10,\n",
+    "    output_action=BatchDeploymentOutputAction.APPEND_ROW,\n",
+    "    output_file_name=\"predictions.csv\",\n",
+    "    retry_settings=BatchRetrySettings(max_retries=3, timeout=30),\n",
+    "    logging_level=\"info\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 5.4.3 Create the deployment\n",
+    "\n",
+    "Using the `MLClient` created earlier, we'll now create the deployment in the workspace. This command will start the deployment creation and return a confirmation response while the deployment creation continues."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ml_client.begin_create_or_update(deployment)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5.5 Test the endpoint with sample data\n",
+    "\n",
+    "Using the MLClient created earlier, we'll get a handle to the endpoint. The endpoint can be invoked using the invoke command with the following parameters:\n",
+    "\n",
+    "- endpoint_name - Name of the endpoint\n",
+    "\n",
+    "- input - Dataset object holding the test dataset\n",
+    "\n",
+    "- deployment_name - Name of the specific deployment to test in an endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "registered_data_asset = ml_client.data.get(name=\"fridge-items-images\", label=\"latest\")\n",
+    "\n",
+    "test_data = Input(type=AssetTypes.URI_FILE, path=registered_data_asset.id)\n",
+    "\n",
+    "# invoke the endpoint for batch scoring job\n",
+    "job = ml_client.batch_endpoints.invoke(\n",
+    "    endpoint_name=batch_endpoint_name,\n",
+    "    input=test_data,\n",
+    "    deployment_name=\"non-mlflow-deployment\",  # name is required as default deployment is not set\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get the details of the job\n",
+    "job_name = job.name\n",
+    "batch_job = ml_client.jobs.get(name=job_name)\n",
+    "print(batch_job.status)\n",
+    "# stream the job logs\n",
+    "ml_client.jobs.stream(name=job_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the output files\n",
```

Comment:
Yes. I am working on it. Basically the output is directly stored in the storage account with a dynamic path. I need to get that dynamic path from the batchDeployment endpoint, which I am unable to get.

Reply:
I was able to download the output using `ml_client.jobs.download` method.
For visualization of the output, I need to use azureml-core, since I am not able to download the images from datastore using azure-ai-ml. I was told that dpv2 is only for control plane operations and for downloading the data I should use something outside of it like azure-storage-sdk. Could you please check if this looks ok and could be used?